release-notes/7.0/7.0.0/api-diff/Microsoft.NETCore.App/7.0.0_System.Runtime.Serialization.DataContracts.md

```diff
@@ -0,0 +1,64 @@
+# System.Runtime.Serialization.DataContracts
+
+``` diff
++namespace System.Runtime.Serialization.DataContracts {
++    public abstract class DataContract {
++        public virtual DataContract? BaseContract { [RequiresUnreferencedCodeAttribute("Data Contract Serialization and Deserialization might require types that cannot be statically analyzed. Make sure all of the required types are preserved.")] get; }
++        public virtual string? ContractType { get; }
++        public virtual ReadOnlyCollection<DataMember> DataMembers { get; }
++        public virtual bool IsBuiltInDataContract { get; }
++        public virtual bool IsISerializable { get; }
++        public virtual bool IsReference { get; }
++        public virtual bool IsValueType { get; }
++        public virtual Dictionary<XmlQualifiedName, DataContract>? KnownDataContracts { [RequiresUnreferencedCodeAttribute("Data Contract Serialization and Deserialization might require types that cannot be statically analyzed. Make sure all of the required types are preserved.")] get; }
++        public virtual Type OriginalUnderlyingType { get; }
++        public virtual XmlDictionaryString? TopLevelElementName { get; }
++        public virtual XmlDictionaryString? TopLevelElementNamespace { get; }
++        public virtual Type UnderlyingType { get; }
```

Comment:
```suggestion
+        [DynamicallyAccessedMembers(ClassDataContract.DataContractPreserveMemberTypes)]
+        public virtual Type UnderlyingType { get; }
```

I don't know if the `DynamicallyAccessedMember` attribute is importatnt for documenting here. It appears the `RequiresUnreferencedCode` attribute is documented. The parameter for the attribute here is an internal constant.

Reply:
I'll add `DynamicallyAccessedMembers` to the list of attributes to ignore.